python/ray/train/lightning/_lightning_utils.py

```diff
@@ -259,9 +259,14 @@ class RayTrainReportCallback(pl.callbacks.Callback):
 
     def __init__(self) -> None:
         super().__init__()
-        self.trial_name = train.get_context().get_trial_name()
+        job_id = ray.get_runtime_context().get_job_id()
+        experiment_name = train.get_context().get_experiment_name()
```

Comment:
we should switch to using run_id in the future, but currently you can't access the run id from the public `ray.train.get_context()`.

job id could technically collide for different tune trials.

Also, I believe lightning has a requirement for all workers on the same node to call `save_checkpoint` on the same directory, which is why we don't just create a random path with uuid here, since that would be different per worker.